core/sr-io/without_std.rs

```diff
@@ -331,8 +331,102 @@ pub mod ext {
 		// Offchain-worker Context
 		//================================
 
-		/// Submit extrinsic.
-		fn ext_submit_extrinsic(data: *const u8, len: u32);
+		/// Submit transaction.
+		///
+		/// Returns 0 if it was successfuly added to the pool, nonzero otherwise.
+		fn ext_submit_transaction(data: *const u8, len: u32) -> u32;
+
+		/// Sign a piece of data with authority key.
+		///
+		/// Returns `0` if successful, nonzero otherwise.
+		fn ext_sign(data: *const u8, len: u32, sig_data: *mut u8) -> u32;
+
+		/// Returns current UNIX timestamp (milliseconds)
+		fn ext_timestamp() -> u64;
+
+		/// Pause execution until `deadline` is reached.
+		fn ext_sleep_until(deadline: u64);
+
+		/// Generate a random seed
+		fn ext_random_seed(data: *mut u8);
+
+		/// Write a value to local storage.
+		fn ext_local_storage_set(key: *const u8, key_len: u32, value: *const u8, value_len: u32);
+
+		/// Read a value from local storage.
+		///
+		/// if len is u32::max_value the value has not been found.
+		fn ext_local_storage_read(key: *const u8, key_len: u32, value_len: *mut u32) -> *mut u8;
+
+		/// Initiaties a http request.
+		///
+		/// If returned value is `u32::max_value` the request couldn't have started.
+		fn ext_http_request_start(
+			method: *const u8,
+			method_len: u8,
```

Comment:
I am not entirely sure why would we want to use `u8` here?

On wasm32 platforms it would be passed as `i32` anyway. I am not entirely convienced that the optimizer can take advantage of `u8` and that it would actually be benefital.

Consider changing it to u32.